yt_dlp/extractor/ntvde.py

```diff
@@ -34,40 +33,36 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         info = self._parse_json(self._search_regex(
-            r'(?s)ntv\.pageInfo\.article\s*=\s*(\{.*?\});', webpage, 'info'),
+            r'(?s)article:\s*(\{.+?\})', webpage, 'info'),
             video_id, transform_source=js_to_json)
         timestamp = int_or_none(info.get('publishedDateAsUnixTimeStamp'))
-        vdata = self._parse_json(self._search_regex(
-            r'(?s)\$\(\s*"\#player"\s*\)\s*\.data\(\s*"player",\s*(\{.*?\})\);',
+
+        player_data = self._parse_json(self._search_regex(
+            r'(?s)\$\(\s*"\#playerwrapper"\s*\)\s*\.data\(\s*"player",\s*(\{.*?\})\);',
             webpage, 'player data'), video_id,
-            transform_source=lambda s: js_to_json(re.sub(r'advertising:\s*{[^}]+},', '', s)))
-        duration = parse_duration(vdata.get('duration'))
+            transform_source=lambda s: js_to_json(re.sub(r'ivw:\s*.+', '', s)))
+        vdata = traverse_obj(player_data, ('setup', 'source'))
+        duration = int_or_none(vdata.get('length'))
 
         formats = []
-        if vdata.get('video'):
-            formats.append({
-                'format_id': 'flash',
-                'url': 'rtmp://fms.n-tv.de/%s' % vdata['video'],
-            })
-        if vdata.get('videoMp4'):
+        if vdata.get('progressive'):
             formats.append({
-                'format_id': 'mobile',
-                'url': compat_urlparse.urljoin('http://video.n-tv.de', vdata['videoMp4']),
-                'tbr': 400,  # estimation
+                'format_id': 'mp4-0',
+                'url': vdata['progressive'],
             })
-        if vdata.get('videoM3u8'):
-            m3u8_url = compat_urlparse.urljoin('http://video.n-tv.de', vdata['videoM3u8'])
+        if vdata.get('hls'):
             formats.extend(self._extract_m3u8_formats(
-                m3u8_url, video_id, ext='mp4', entry_protocol='m3u8_native',
+                vdata['hls'], video_id, ext='mp4', entry_protocol='m3u8_native',
                 quality=1, m3u8_id='hls', fatal=False))
-
+        if vdata.get('dash'):
+            formats.extend(self._extract_mpd_formats(vdata['dash'], video_id, fatal=False))
         return {
             'id': video_id,
             'title': info['headline'],
             'description': info.get('intro'),
             'alt_title': info.get('kicker'),
             'timestamp': timestamp,
-            'thumbnail': vdata.get('html5VideoPoster'),
+            'thumbnail': vdata.get('poster'),
             'duration': duration,
             'formats': formats,
         }
```

Comment:
```python
return {
    'id': video_id,
    **traverse_obj(info, {
        'title': 'headline',
        'description': 'intro',
        'alt_title': 'kicker',
        'timestamp': ('publishedDateAsUnixTimeStamp', {int_or_none}),
    }),
    **traverse_obj(vdata, {
        'thumbnail': 'poster',
        'duration': ('length', {int_or_none}),
    }),
    'formats': formats,
}
```

Reply:
Same as above: My Python knowledge is very limited, I didn't know you could use `traverse_obj()` this way :)

Anyway, it works fine with the changes.